PerimeterXModule/Internals/Validators/PXCookieValidator.cs

```diff
@@ -1,33 +1,69 @@
 ﻿using System;
 using System.Diagnostics;
+using System.Text.RegularExpressions;
 
 namespace PerimeterX
 {
-	class PXCookieValidator : IPXCookieValidator
+	public class PXCookieValidator : IPXCookieValidator
 	{
-		private PxModuleConfigurationSection config;
+		public static readonly string CALL_REASON_INVALID_COOKIE = "invalid_cookie";
+		public static readonly string CALL_REASON_DECRYPTION_FAILED = "cookie_decryption_failed";
+		public static readonly string CALL_REASON_VALIDATION_FAILED = "cookie_validation_failed";
+		public static string CALL_REASON_NO_COOKIE = "no_cookie";
+		public static string CALL_REASON_EXPIRED_COOKIE = "cookie_expired";
+		public static string CALL_REASON_SENSITIVE_ROUTE = "sensitive_route";
+		public static string CALL_REASON_MOBILE_ERROR = "mobile_error_{0}";
+		public IPXCookieValidator PXOriginalCookieValidator { get; set; }
+
+		protected PxModuleConfigurationSection config;
+		private readonly string MOBILE_PATTERN_ERROR = @"^\d+$";
 
 		public PXCookieValidator(PxModuleConfigurationSection config)
 		{
 			this.config = config;
 		}
 
-		public bool CookieVerify(PxContext context, IPxCookie pxCookie)
+		public virtual bool Verify(PxContext context, IPxCookie pxCookie)
 		{
 			try
 			{
 				if (pxCookie == null)
 				{
 					Debug.WriteLine("Request without risk cookie - " + context.Uri, PxConstants.LOG_CATEGORY);
-					context.S2SCallReason = RiskRequestReasonEnum.NO_COOKIE;
+					context.S2SCallReason = CALL_REASON_NO_COOKIE;
 					return false;
 				}
 
+				if (context.CookieOrigin.Equals(CookieOrigin.HEADER))
+				{
+					string authorizatoinHeader = context.GetHeadersAsDictionary()[PxConstants.MOBILE_HEADER.ToLower()];
+					Regex match = new Regex(MOBILE_PATTERN_ERROR);
+					if (!string.IsNullOrEmpty(authorizatoinHeader) && match.IsMatch(authorizatoinHeader))
```

Comment:
You can apply the regex inline, and then line 40 can be removed - 
`if (.. && Regex.Match(authorizatoinHeader, MOBILE_PATTERN_ERROR).Success)`

Reply:
Perhaps it is worth to add a debug message here (the case of identifying a mobile sdk error).